src/python/TaskWorker/Actions/PostJob.py

```diff
@@ -2607,8 +2622,43 @@ def set_dashboard_state(self, state, reason = None, exitCode = None):
                  }
         if reason:
             params['StatusValueReason'] = reason
-        ## Take exit code from job_fjr.<job_id>.<retry_id>.json and report final exit code to Dashboard.
-        ## Only taken if RetryJob think it is FATAL_ERROR.
+        self.monitoringExitCode(params, exitCode)
+        ## Unfortunately, Dashboard only has 1-second resolution; we must separate all
+        ## updates by at least 1s in order for it to get the correct ordering.
+        time.sleep(1)
+        msg += " Dashboard parameters: %s" % (str(params))
+        self.logger.info(msg)
+        DashboardAPI.apmonSend(params['MonitorID'], params['MonitorJobID'], params)
+        self.logger.info("====== Finished to prepare/send report to dashboard.")
+
+    ## = = = = = PostJob = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
+
+    def set_postJob_ClassAds(self, state, exitCode = None):
+        """
+        Update PostJobStatus and job exit-code among the job ClassAds for the monitoring script to update the Grafana dashboard.
+        """
+        if os.environ.get('TEST_POSTJOB_NO_STATUS_UPDATE', False):
+            return
+        msg = "postJob status: %s." % (state)
+        params = {'PostJobStatus': state}
+        self.monitoringExitCode(params, exitCode)
+        msg += " ClassAds values to set are: %s" % (str(params))
+        self.logger.info(msg)
+        schedd = htcondor.Schedd()
+        with HTCondorUtils.AuthenticatedSubprocess(os.environ['X509_USER_PROXY'], logger=self.logger) as (parent, rpipe):
+            if not parent:
+                for param in params:
+                    schedd.edit([self.dag_jobid], param, "'"+params[param]+"'")
+                # Allow Condor to remove the job from the queue before ten days upon completion
```

Comment:
I fully do not understand this comment. English is somehow broken but mostly.. which ten days ? I do not see a ref. to ten days.

Reply:
See the `LeaveJobInQueue` description here:
https://research.cs.wisc.edu/htcondor/manual/v7.6/10_Appendix_A.html

I agree the English is not clear :)

Reply:
I do not think that we use condor_submit -spool  In any case we want to define and enforce our own policy. 10 days are clearly too much.